harness/determined/common/api/authentication.py

```diff
@@ -97,6 +97,7 @@ def _init_session(
 
         if (
             token is None
+            and requested_user in (None, util.get_det_username_from_env())
```

Comment:
Would you mind commenting what some of these cases are for?

It's getting really hard to reverse-engineer the intention from the code.

At a very minimum, please comment what the case you are adding is for.  Don't be afraid to write a long, clear, precise comment... it will help somebody someday.

Reply:
Done!